### PR TITLE
Remove old 1.x auth stack

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/aspnet/AspNetCore/issues/3999

@Tratcher

### CODE REVIEW

**Review by Tratcher — CHANGES_REQUESTED** [review]
All of the servers implemented IHttpAuthenticationFeature, you'll need to remove the property there as well.

**Comment by HaoK** [comment]
Updated HttpSys and IIS to remove the Handler property

**Comment by Tratcher** [comment]
And kestrel?

**Comment by HaoK** [comment]
I couldn't see anything obvious that needed updating in kestrel and it didn't fail in building...  PR validation looks like it failed just due to breaking changes for the removal, fixing those now

**Comment by Tratcher** [comment]
I guess in Kestrel it's only stored in the collection feature collection, it's not actually implemented. https://github.com/aspnet/KestrelHttpServer/search?q=IHttpAuthenticationFeature&unscoped_q=IHttpAuthenticationFeature

**Review by Tratcher — APPROVED** [review]

File: src/Http/Http.Features/src/breakingchanges.netcore.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+  [
+    {
+      "TypeId": "public class Microsoft.AspNetCore.Http.Features.Authentication.AuthenticateContext",
+      "Kind": "Removal"
+    },
+    {
+      "TypeId": "public class Microsoft.AspNetCore.Http.Features.Authentication.ChallengeContext",
+    },
+      "Kind": "Removal"
```
Comment (PinpointTownes) [inline src/Http/Http.Features/src/breakingchanges.netcore.json]: Invalid JSON?

**Comment by PinpointTownes** [comment]
:+1: that was an incredible source of pain for users not familiar with the Auth 2.0 "grand refactoring".

**Comment by Tratcher** [comment]
```
2018-12-12T00:52:00.4744658Z   D:\a\1\s\src\Security\src\Microsoft.AspNetCore.Authentication.JwtBearer\JwtBearerOptions.cs(103,24): error CS1574: XML comment has cref attribute 'AuthenticationProperties' that could not be resolved [D:\a\1\s\src\Security\src\Microsoft.AspNetCore.Authentication.JwtBearer\Microsoft.AspNetCore.Authentication.JwtBearer.csproj]
2018-12-12T00:52:00.4745720Z   D:\a\1\s\src\Security\src\Microsoft.AspNetCore.Authentication.Twitter\Messages\RequestToken.cs(4,33): error CS0234: The type or namespace name 'Authentication' does not exist in the namespace 'Microsoft.AspNetCore.Http' (are you missing an assembly reference?) [D:\a\1\s\src\Security\src\Microsoft.AspNetCore.Authentication.Twitter\Microsoft.AspNetCore.Authentication.Twitter.csproj]
2018-12-12T00:52:00.4905085Z   D:\a\1\s\src\Security\src\Microsoft.AspNetCore.Authentication.Twitter\Messages\RequestTokenSerializer.cs(6,33): error CS0234: The type or namespace name 'Authentication' does not exist in the namespace 'Microsoft.AspNetCore.Http' (are you missing an assembly reference?) [D:\a\1\s\src\Security\src\Microsoft.AspNetCore.Authentication.Twitter\Microsoft.AspNetCore.Authentication.Twitter.csproj]
2
```

**Comment by HaoK** [comment]
@natemcmaster are the test failures expected at this point, pr-validation-temp passed, is this PR ok to merge?

**Comment by natemcmaster** [comment]
Depends on which tests failed and why. I recommend comparing your PR with https://github.com/aspnet/AspNetCore-Internal/labels/test-failure to see if failures are known or not.

**Comment by HaoK** [comment]
This PR is just removing obsolete apis so its either a compilation break or no change, the failures mostly appear to be in kestrel so should be fine

**Comment by Tratcher** [comment]
The IIS failures look related.
https://dnceng.visualstudio.com/public/_build/results?buildId=57950&view=logs&jobId=60ad5fa3-9ee1-561c-fc1d-1349a7b472eb&taskId=71e748b8-0ded-5502-f62f-08e9d9d9dfd7&lineStart=697&lineEnd=698&colStart=1&colEnd=1

IIS is still using package references. @natemcmaster would he need to check in and then update the IIS packages in the next build?

**Comment by natemcmaster** [comment]
We are close to getting IIS onto project ref, so I would be okay ignoring the IIS test failures for a day. Okay with you @pakrym ?

**Comment by pakrym** [comment]
As long as we don't forget about them I'm fine.

**Comment by HaoK** [comment]
I'm happy to wait to merge next week as well, whatever is easiest for you guys is fine, this PR is done at this point so I can wait to merge when convenient

**Comment by davidfowl** [comment]
@HaoK my bad bro, I broke all of your stuff

**Comment by HaoK** [comment]
All good, rebased against master, will try to merge today if the checks pass

**Comment by HaoK** [comment]
Wow checks all passed after rebasing and resolving conflicts, so safe to click the green button here @natemcmaster @Tratcher ?

**Review by davidfowl — APPROVED** [review]
Wow green! LGTM

**Comment by Tratcher** [comment]
Yup, go for it.

**Pull request merged**